packages/react-instantsearch-core/src/connectors/connectHitsInsights.ts

```diff
@@ -0,0 +1,77 @@
+import createConnector from '../core/createConnector';
+import { getResults } from '../core/indexUtils';
+
+type Without<T, K> = Pick<T, Exclude<keyof T, K>>;
+type Results = { index: string };
+type Hit = { objectID: string; __position: number; __queryID: string };
+
+type InsightsClient = (
+  method: InsightsClientMethod,
+  payload: InsightsClientPayload
+) => void;
+
+type InsightsClientMethod =
+  | 'clickedObjectIDsAfterSearch'
+  | 'convertedObjectIDsAfterSearch';
+
+type InsightsClientPayload = {
+  index: string;
+  queryID: string;
+  eventName: string;
+  objectIDs: string[];
+  positions?: number[];
+};
+
+function inferPayload({
+  method,
+  results,
+  currentHit,
+}: {
+  method: InsightsClientMethod;
+  results: Results;
+  currentHit: Hit;
+}): Without<InsightsClientPayload, 'eventName'> {
+  const { index } = results;
+  const queryID = currentHit.__queryID;
+  const objectIDs = [currentHit.objectID];
+  switch (method) {
+    case 'clickedObjectIDsAfterSearch': {
+      const positions = [currentHit.__position];
+      return { index, queryID, objectIDs, positions };
+    }
+
+    case 'convertedObjectIDsAfterSearch':
+      return { index, queryID, objectIDs };
+
+    default:
+      throw new Error(
+        `Unsupported method "${method}" passed to the insights function. The supported methods are: "clickedObjectIDsAfterSearch", "convertedObjectIDsAfterSearch".`
+      );
+  }
+}
+
+const wrapInsightsClient = (
+  aa: InsightsClient,
+  results: Results,
+  currentHit: Hit
+) => (
+  method: InsightsClientMethod,
+  payload: Partial<InsightsClientPayload>
+) => {
+  if (typeof aa !== 'function') {
+    throw new TypeError(`Expected insightsClient to be a Function`);
+  }
+  const inferredPayload = inferPayload({ method, results, currentHit });
+  aa(method, { ...inferredPayload, ...payload } as any);
+};
+
+export default (insightsClient: InsightsClient) =>
```

Comment:
Why do we inject it directly at this level rather than on `InstantSearch` like we do for [`searchClient`](https://github.com/algolia/react-instantsearch/blob/34dbb68768407c38f2f50b2b06272e29f1847677/packages/react-instantsearch-core/src/core/createInstantSearch.js#L64-L69)? It might leads to some complication if we have to perform some global configuration on the client (like UA). What happen if we use it at three different place? We add the UA three times?

Reply:
Good point, I don't know if we can change the client from instantsearch since it can be used outside as well in the same application.
Currently it is not possible to have multiple insights clients running at the same time. https://github.com/algolia/search-insights.js/issues/67

Reply:
With discuss about that live, it's fine for the moment. This API is not intrusive at all which means that we can throw it away if at some point we take the decision to have a deeper integration of insights inside the library.